File: code/validation/validate_filenames.R

# Validate forecast filename equals forecast filepath
check_forecast_name_path <- function(forecast_file) {
    forecast_file_path <- basename(dirname(forecast_file))
    forecast_file_name_base <-  substring(basename(forecast_file),
                                        12,
                                        nchar(basename(forecast_file)) - 4)

    if (forecast_file_path != forecast_file_name){
        error_message <- paste("\nERROR: Forecast file name: ",
                               basename(forecast_file),
                                " does not match Forecast file naming convention: ",
                                "<date>-<team>-<model>.csv"
                               )
        return(stop(error_message))
    }
    else{
        check_accepted <- paste("✔ Forecast file name = Forecast file path (",
                                    forecast_file_name_base,
                                    "=",
                                    forecast_file_path,
                                    ")")
        return(writeLines(check_accepted))
    }
    return(0)
}

# VERIFY forecasts filenames
forecasts <- list.files(path="./data-processed", pattern="*.csv", full.names=TRUE, recursive=TRUE)
for(i in forecasts){
    writeLines(paste("\nTesting", i, "..."))
    # check if forecast file path = forecast file name
    check_forecast_name_path(i)
}

File: code/validation/validate_filenames.py
import sys
from pathlib import Path

p = Path('./data-processed/')

# all folders in data-processed
folders = [f for f in p.iterdir()]

for folder in folders:
    # all .csv files the given folder
    file_names = [f for f in folder.iterdir() if f.name.endswith('.csv')]

    # remove ending '.csv', remove date, split between country and model name
    model_names = [f.stem[11:] for f in file_names]

    # for each file check if the model name equals the folder name
    for i, m in enumerate(model_names):
        print('\nTesting ' + file_names[i].name + '...')
        if folder.name == m:
            print('✔ Forecast file name = Forecast file path (' +
                 m + ' = ' + folder.name  + ')')
        else:
            error_message = ("\nERROR: Forecast file name: " + file_names[i].name +
                        " does not match forecast file naming convention: " +
                        "<date>-<team>-<model>.csv")
            sys.exit(error_message)

